analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/CopyPasteTokenAnalyzerTest.cs

```diff
@@ -112,13 +107,28 @@ public void Verify_Duplicated_CS_GlobalUsings()
         public void Verify_NotRunForTestProject_CS()
         {
             const string testRoot = BasePath + nameof(Verify_NotRunForTestProject_CS);
-            builder
-                .AddAnalyzer(() => new TestCopyPasteTokenAnalyzer_CS(testRoot, true))
-                .AddPaths("DuplicatedDifferentLiterals.cs")
-                .WithProtobufPath(@$"{testRoot}\token-cpd.pb")
+            CreateBuilder(new TestCopyPasteTokenAnalyzer_CS(testRoot, true), "DuplicatedDifferentLiterals.cs", AnalyzerLanguage.CSharp, testRoot)
                 .VerifyUtilityAnalyzerProducesEmptyProtobuf();
         }
 
+        [DataTestMethod]
+        [DataRow("Unique.cs", true)]
+        [DataRow("SomethingElse.cs", false)]
+        public void Verify_UnchangedFiles(string unchangedFileName, bool expectedProtobufIsEmpty)
+        {
+            var testRoot = BasePath + nameof(Verify_UnchangedFiles);
+            var builder = CreateBuilder(new TestCopyPasteTokenAnalyzer_CS(testRoot, false), "Unique.cs", AnalyzerLanguage.CSharp, testRoot)
+                .WithSonarProjectConfigPath(TestHelper.CreateSonarProjectConfig(TestContext, new[] { BasePath + unchangedFileName }));
+            if (expectedProtobufIsEmpty)
+            {
+                builder.VerifyUtilityAnalyzerProducesEmptyProtobuf();
+            }
+            else
+            {
+                builder.VerifyUtilityAnalyzer<TokenTypeInfo>(x => x.Should().NotBeEmpty());
+            }
```

Comment:
In this area, it's very easy to write a test that will "not fail" for the wrong reason, due to some scaffolding issues. That's why I always made it parametrized, to make sure that scaffolding with one parameter works end-to-end, and the same scaffolding with different value produces different result.

I faced several times that I didn't scaffold something properly and the test passed. Consider the non-empty variant - we can remove the "WithSonarProjectConfigPath` part completely, and it will still be green.